src/components/verification-grid/verification-grid.ts

```diff
@@ -292,6 +338,194 @@ export class VerificationGridComponent extends AbstractComponent(LitElement) {
+      const matchingAspectRatio = 16 / 9;
+      const matchingDeviation = 0.2;
+
+      if (Math.abs(aspectRatio - matchingAspectRatio) <= matchingDeviation) {
+        return { columns: 2, rows: 2 };
+      }
+    }
+
+    const containerSize = this.gridContainer.getBoundingClientRect();
+    let refinedTarget = target;
+    const testedTargets: number[] = [];
+
+    // some numbers e.g. 9 have one factor such as 3x3 which creates a aspect
+    // ratio of 1.00
+    // using a 3x3 grid for a 16:9 screen is not ideal because it will create
+    // a lot of dead space
+    // therefore, we have a threshold that we have to meet. If we do not meet
+    // the threshold, we keep increasing the target until we find a grid shape
+    // that meets the similarity percentage threshold
+    const targetThreshold = 0.5;
+
+    // eslint-disable-next-line no-constant-condition
+    while (true) {
+      const candidateShapes: GridShape[] = [];
+
+      // sometimes we can get stuck in an infinite loop where increasing grid
+      // size will result in the grid not fitting, but decreasing the grid size
+      // will result in a grid that does not meet the aspect ratio threshold
+      // to resolve this, we keep a record of the targets that we have checked
+      // if we see the same one twice (we have increased, then decreased)
+      // we will decrement the grid size until we find a grid that fits
+      if (testedTargets.includes(refinedTarget)) {
+        refinedTarget -= 1;
+        continue;
+      }
+      testedTargets.push(refinedTarget);
+
+      let hasFittingCandidate = false;
+      for (let i = 1; i <= refinedTarget; i++) {
+        if (refinedTarget % i === 0) {
+          const columns = i;
+          const rows = refinedTarget / i;
+
+          const candidateShape = { columns, rows };
+          const willFit = this.willFitTileSize(containerSize, candidateShape);
+          const meetsAspectRatio = this.gridAspectRatioSimilarity(containerSize, candidateShape) > targetThreshold;
+
+          if (willFit) {
+            hasFittingCandidate = true;
+
+            if (meetsAspectRatio) {
+              candidateShapes.push({ columns, rows });
+            }
+          }
+        }
+      }
+
+      // if we have reached this condition, there is a fatal bug in the
+      // dynamic grid size code. This condition should never be triggered
+      // I have added the condition so that if dynamic grid size gets stuck in
+      // and infinite loop it won't completely crash the browser
+      // meaning that the verification grid will soft fail to prevent all the
+      // users work being lost
+      const maxGridSize = 1_000 as const;
+      if (refinedTarget >= maxGridSize || refinedTarget <= 0) {
+        console.error("Reached maximum/minimum grid size. Recovering via a 1x1 grid.");
+        this.targetGridSize = 1;
+        return { columns: 1, rows: 1 };
+      }
```

Comment:
basically: this needs to go. this algorithm is either sufficiently tested that this case never occurs, or it never occurs and it is useless code.